### PR TITLE
Add MergeStrategy to objectArrayTrait

### LINKED ISSUE #6882: The `y-column` attr in `FeatureInfoPanelChart` (`<chart>`) is not working
Related to 

- #6434 
- https://github.com/TerriaJS/terriajs/issues/6758
- https://github.com/TerriaJS/terriajs/issues/2258

Example - the feature info template is supposed to show `z` column along x-axis and the `plant height` column along y-axis.

```json
    {
      "name": "Test",
      "type": "csv",
      "url": "http://ci.terria.io/main/build/TerriaJS/test/csv/lat_lon_nullvalue.csv",
      "featureInfoTemplate": {
        "name": "Number {{value}}",
        "template": "<p>All train stations have the same x-y plot.</p><chart src='http://ci.terria.io/main/build/TerriaJS/test/csv_nongeo/x_height.csv' x-column='z' y-columns='plant height'></chart>"
      }
    }
```

Instead it uses `z` as the `yColumn`

- [Test link](http://ci.terria.io/main/#clean&start=%7B%22version%22%3A%20%228.0.0%22%2C%20%22initSources%22%3A%20%5B%7B%22catalog%22%3A%20%5B%7B%22name%22%3A%20%22Train%20Stations%22%2C%22type%22%3A%20%22csv%22%2C%22url%22%3A%20%22http%3A%2F%2Fci.terria.io%2Fmain%2Fbuild%2FTerriaJS%2Ftest%2Fcsv%2Flat_lon_nullvalue.csv%22%2C%22featureInfoTemplate%22%3A%20%7B%22name%22%3A%20%22Number%20%7B%7Bvalue%7D%7D%22%2C%22template%22%3A%20%22%3Cp%3EAll%20train%20stations%20have%20the%20same%20x-y%20plot.%3C%2Fp%3E%3Cchart%20src%3D'http%3A%2F%2Fci.terria.io%2Fmain%2Fbuild%2FTerriaJS%2Ftest%2Fcsv_nongeo%2Fx_height.csv'%20x-column%3D'z'%20y-columns%3D'plant%20height'%3E%3C%2Fchart%3E%22%7D%7D%5D%7D%5D%7D)
- Add "Test" to workbench
- Click on feature
- It looks like

image

It should look something like

image

### PR DESCRIPTION
**Pull request by nf-s**
### What this PR does

Fixes https://github.com/TerriaJS/terriajs/issues/6882

- Add `MergeStrategy` to `objectArrayTrait` - this includes a new `topStratum` strategy - similar to `Merge.All` (the default behaviour), but only elements that exist in the top-most strata will be merged with lower strata. Elements that only exist in lower strata will be removed.
- Fix `y-column` in `FeatureInfoPanelChart` (`<chart>`)

### Test me

For example context see #6882:
* [main](http://ci.terria.io/main/#clean&https://gist.githubusercontent.com/steve9164/fa3a23c39ad8c351eab7604f83ec4e7f/raw/d38747e720513d7f9e61c0d736d48eecb1396666/test-fi-chart.json)
* [array-merge-top](http://ci.terria.io/array-merge-top/#clean&https://gist.githubusercontent.com/steve9164/fa3a23c39ad8c351eab7604f83ec4e7f/raw/d38747e720513d7f9e61c0d736d48eecb1396666/test-fi-chart.json)
* Open three catalog items, click on feature, and look at chart (you can expand the chart to confirm correct columns)

The feature info chart plots:

```
x,z,plant height
1,-1,2
3,-3,4
5,-5,6
7,-7,3
10,-10,9
12,-12,2
```

`main` always chooses `z` for the y axis

### Checklist

- [x] There are unit tests to verify my changes are correct or unit tests aren't applicable (if so, write quick reason why unit tests don't exist)
- [ ] I've updated relevant documentation in `doc/`.
- [x] I've updated CHANGES.md with what I changed.
- [x] I've provided instructions in the PR description on how to test this PR.

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -3,6 +3,8 @@
 #### next release (8.5.2)
 
 - Add `usePreCachedTilesIfAvailable` to `ArcGisMapServerCatalogItemTraits`.
+- Add `MergeStrategy` to `objectArrayTrait` - this includes a new `topStratum` strategy - similar to `Merge.All` (the default behaviour), but only elements that exist in the top-most strata will be merged with lower strata. Elements that only exist in lower strata will be removed.
```
Comment (tephenavies) [inline CHANGES.md]: I think changing some of the traits to use these merge strategies should be listed as a breaking change.

Comment (nf-s) [reply CHANGES.md]: Good point, I hath added it as a breaking change in CHANGES

File: lib/Traits/Decorators/objectArrayTrait.ts [hunk 2]
```diff
@@ -18,15 +18,26 @@ import ModelTraits from "../ModelTraits";
 import Trait, { TraitOptions } from "../Trait";
 import traitsClassToModelClass from "../traitsClassToModelClass";
 
+export enum MergeStrategy {
+  /**
+   * Merge array elements across strata (if merge is false, each element will only be the top-most strata's object). This is the default.
```
Comment (na9da) [inline lib/Traits/Decorators/objectArrayTrait.ts]: `(if merge is false, each element will only be the top-most strata's object)`
I think this comment should be revised as `if merge strategy is None`.

**Review by na9da — APPROVED** [review]
Looks good @nf-s.

Just one small comment.

**Pull request merged**